### PR TITLE
Add gateway endpoint and Enums for Event class

### PR DESCRIPTION
**Pull request by iMicknl**

### CODE REVIEW

File: pyhoma/models.py [hunk 1]
```diff
@@ -277,3 +389,245 @@ class DataType(Enum):
...
+    UNKNOWN_DETAILED_ERROR = 10007
+
+
+class UpdateBoxStatus(Enum):
+    NOT_UPDATABLE = "NOT_UPDATABLE"
+    READY_TO_UPDATE = "READY_TO_UPDATE"
+    READY_TO_BE_UPDATED_BY_SERVER = "READY_TO_BE_UPDATED_BY_SERVER"
+    READY_TO_UPDATE_LOCALLY = "READY_TO_UPDATE_LOCALLY"
+    UP_TO_DATE = "UP_TO_DATE"
+    UNKNOWN = "UNKNOWN"
```
Comment (tetienne) [inline pyhoma/models.py:633]: It's nice to have these enums. But where should we use them?

Comment (iMicknl) [reply pyhoma/models.py:633]: These enums are used by `Gateway` and could be used to check if there is an update available. Some of the other enums are not for Gateway and need to be mapped in future PR's. I just wanted to include them already before I lose the source again :D.

Comment (iMicknl) [reply pyhoma/models.py:661]: @tetienne included a few already. Would make our coordinator code a bit more strongly typed using those enums.

Comment (tetienne) [reply pyhoma/models.py:661]: That's great!

File: pyhoma/models.py [hunk 2]
```diff
@@ -197,30 +198,81 @@ class Event:
...
+        deleted_raw_devices_count: Optional[Any] = None,
+        protocol_type: Optional[Any] = None,
         gateway_id: Optional[str] = None,
         exec_id: Optional[str] = None,
         deviceurl: Optional[str] = None,
         device_states: Optional[List[Dict[str, Any]]] = None,
-        old_state: Optional[str] = None,
-        new_state: Optional[str] = None,
+        old_state: Optional[ExecutionState] = None,
+        new_state: Optional[ExecutionState] = None,
```
Comment (tetienne) [inline pyhoma/models.py:242]: So many cases possible :o

File: pyhoma/models.py (hunk 1 above)
Comment (tetienne) [reply pyhoma/models.py:745]: This class is now really huge. What do you think about moving all the enums to a dedicated module?

Comment (iMicknl) [reply pyhoma/models.py:745]: > This class is now really huge. What do you think about moving all the enums to a dedicated module?

Should I create an `enums.py`?

**Review by tetienne — APPROVED** [review]

**Pull request merged**